### PR TITLE
wip: fix types in train-stream

### PR DESCRIPTION
**Pull request by mubaidr**
Update/Fix types in train-steam module

### CODE REVIEW

File: src/train-stream.ts [hunk 1]
```diff
@@ -24,19 +29,17 @@ export class TrainStream extends Writable {
...
   iterations: number;
   errorThresh: number;
-  log: (message: string) => void;
+  log: boolean | ((status: INeuralNetworkState) => void);
   logPeriod: number;
   callbackPeriod: number;
-  callback: (stats: { error: number; iterations: number }) => void;
-  firstDatum: any;
+  callback?: (status: { iterations: number; error: number }) => void;
+  firstDatum: INeuralNetworkDatum[] | undefined;
```
Comment (mubaidr) [inline src/train-stream.ts]: @robertleeplummerjr Please confirm types for `firstDatum` and `chunk`

File: src/train-stream.ts [hunk 2]
```diff
@@ -92,16 +96,23 @@ export class TrainStream extends Writable {
...
+
       return next();
     }
 
     this.count++;
 
     const data = this.neuralNetwork.formatData(chunk);
-    // TODO: Remove this typecast once neural network classes are typed
-    this.sum += this.neuralNetwork.trainPattern(data[0], true) as number;
+    const error = this.neuralNetwork.trainPattern(data[0], true);
```
Comment (mubaidr) [inline src/train-stream.ts]: @robertleeplummerjr Also not sure about this one either.

**Comment by robertleeplummerjr** [comment]
Very good job!

**Review by robertleeplummerjr — APPROVED** [review]
Awesome work.

**Pull request merged**